dev_docs/tutorials/testing_plugins.mdx

```diff
@@ -0,0 +1,1544 @@
+          id: 'none',
+          name: 'none',
+        }
+      }
+    });
+  });
+
+  it('handles undefined', () => {
+    expect(migrateCaseFromV7_9_0ToV7_10_0({
+      id: '1',
+      type: 'case',
+      attributes: {
+        // Even though undefined isn't a valid JSON or Elasticsearch value, we should test it anyways since there
+        // could be some JavaScript layer that casts the field to `undefined` for some reason.
+        connector_id: undefined 
+      }
+    })).toEqual({
+      id: '1',
+      type: 'case',
+      attributes: {
+        connector: {
+          id: 'none',
+          name: 'none',
+        }
+      }
+    });
+
+    expect(migrateCaseFromV7_9_0ToV7_10_0({
+      id: '1',
+      type: 'case',
+      attributes: {
+        // also test without the field present at all
+      }
+    })).toEqual({
+      id: '1',
+      type: 'case',
+      attributes: {
+        connector: {
+          id: 'none',
+          name: 'none',
+        }
+      }
+    });
+  });
+});
+```
+
+#### Integration testing
+With more complicated migrations, the behavior of the migration may be dependent on values from other plugins which may
+be difficult or even impossible to test with unit tests. You need to actually bootstrap Kibana, load the plugins, and
+then test the full end-to-end migration. This type of set up will also test ingesting your documents into Elasticsearch
+against the mappings defined by your Saved Object type.
+
+This can be achieved using the `jest_integration` suite and the `kbnTestServer` utility for starting an in-memory
+instance of Kibana. You can then leverage the import API to test migrations. This API applies the same migrations to
+imported documents as are applied at Kibana startup and is much easier to work with for testing.
+
+```ts
+import type supertest from 'supertest';
+// You may need to adjust these paths depending on where your test file is located.
+// The absolute path is src/core/test_helpers/kbn_server
+import * as kbnTestServer from '../../../../src/core/test_helpers/kbn_server';
+// The absolute path is src/core/server
+import { Root } from '../../../src/core/server';
+
+describe('my plugin migrations', () => {
+  let esServer: kbnTestServer.TestElasticsearchUtils;
+  let kibanaRoot: Root;
+  let importApi: supertest.Test;
+  let exportApi: supertest.Test;
+
+  beforeAll(async () => {
+    const { startES } = kbnTestServer.createTestServers({ adjustTimeout: jest.setTimeout });
+    esServer = await startES();
+    root = kbnTestServer.createRootWithCorePlugins({}, { oss: false }) // oss: false flag necessary if you need to load x-pack plugins
+    await root.setup();
+    await root.start();
+    // Create a pre-configured instance of supertest for the import export APIs
+    importApi = kbnTestServer.getSupertest(root, 'POST', '/api/saved_objects/_import').set('Content-Type: application/ndjson');
+    exportApi = kbnTestServer.getSupertest(root, 'POST', '/api/saved_objects/_export')
```

Comment:
This is a pretty complicated set up for developers to manage. I'm going to attempt adding a simple utility to wrap this functionality to this PR

Reply:
Opened a new PR for this #106116